aim/cli/up/commands.py

```diff
@@ -115,6 +118,76 @@ def up(dev, host, port, workers, repo, tf_logs, ssl_keyfile, ssl_certfile, base_
     click.echo('Press Ctrl+C to exit')
 
     try:
+        if tf_logs:
+            tf_logs_cache_path = f'{repo_inst.path}/tf_logs_cache.json'
+            try:
+                with open(tf_logs_cache_path) as FS:
+                    tf_logs_cache = json.load(FS)
+            except Exception:
+                tf_logs_cache = {}
+
+            for run_path in os.listdir(tf_logs):
+                path = os.path.join(tf_logs, run_path)
+                run_hash = tf_logs_cache.get(path)
+                if run_hash:
+                    run = Run(run_hash=run_hash)
+                    run_tf_hashsums = run['tf_hashsums']
+                else:
+                    run_tf_hashsums = {}
+                    run = Run()
+                    run['tf_hashsums'] = run_tf_hashsums
+                    tf_logs_cache[path] = run.hash
+
+                events = [
+                    os.path.join(root, file)
+                    for root, dirs, files in os.walk(path)
+                    for file in files if file.endswith('v2')
+                ]
+
+                events_to_process = []
+                for event in events:
+                    # Calculate file hash
+                    chunk_size = 8192
+                    file_hash = hashlib.md5()
+                    with open(event, 'rb') as FS:
+                        chunk = FS.read(chunk_size)
+                        while chunk:
+                            file_hash.update(chunk)
+                            chunk = FS.read(chunk_size)
+                    hashsum = file_hash.hexdigest()
+
+                    if hashsum != run_tf_hashsums.get(event, ''):
+                        # Something has changed or hasn't been processed before
+                        events_to_process.append(event)
+                    run_tf_hashsums[event] = hashsum
+                run['tf_hashsums'] = run_tf_hashsums
+
+                if events_to_process:
+                    # This import statement takes long to complete
+                    import tensorflow as tf
+                    from tensorflow.python.framework import tensor_util
+                    from tensorflow.python.summary.summary_iterator import summary_iterator
+
+                    for event in events_to_process:
+                        for e in summary_iterator(event):
+                            timestamp = e.wall_time
+                            step = e.step
+                            for v in e.summary.value:
+                                tag = v.tag
+                                plugin_name = v.metadata.plugin_data.plugin_name
+                                if plugin_name == 'images':
```

Comment:
can we check the `tensor` shape and if it's a 4-D tensor, treat it as an image list?

Reply:
Cannot seem to resolve this,